MetaMask/metamask-extension
# fix: blockFilter flaky rpc test

## **Description**
This fixes a race condition with the `filterChanges` and `blockByNumber` requests where a block could occur between them.


[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/24881?quickstart=1)

## **Related issues**

Fixes: #24625

## **Manual testing steps**

1. `yarn build:test`
2. introduce a `await driver.delay(9000);` between `eth_getBlockByNumber` and `eth_getFilterChanges`. 
3. run `yarn test:e2e:single --browser chrome --leave-running test/e2e/json-rpc/eth_newBlockFilter.spec.js`
4. see that it passes (it would not pass on develop)

## **Screenshots/Recordings**

![image](https://github.com/MetaMask/metamask-extension/assets/364566/7af0ffa7-256c-4179-b207-661033bf7286)



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
